includes/Pager.php

```diff
@@ -284,20 +289,22 @@ function getSqlComment() {
 	 * @param $descending Boolean: query direction, false for ascending, true for descending
 	 * @return ResultWrapper
 	 */
-	function reallyDoQuery( $offset, $limit, $descending ) {
+	function reallyDoQuery( $offset, $limit, $descending, $info = [] ) {
 		$fname = __METHOD__ . ' (' . $this->getSqlComment() . ')';
-		$info = $this->getQueryInfo();
+		if ( !isset( $info ) ) {
+			$info = $this->getQueryInfo();
```

Comment:
Why is standard subclassing & implementing of `abstract` function `getQueryInfo` not viable here?

Reply:
In order to add $info as a parameter we need this if to override only default value of $info which is an empty array, otherwise, it should use $info from the argument.